apps/explorer/priv/repo/migrations/20221214162647_lock_stats_table.exs

```diff
@@ -0,0 +1,64 @@
+defmodule Explorer.Repo.Local.Migrations.LockStatsTable do
+  use Ecto.Migration
+
+  def up do
+    execute("""
+    CREATE OR REPLACE FUNCTION celo_transaction_stats_trigger_func() RETURNS trigger
+    LANGUAGE plpgsql AS
+    $$BEGIN
+
+    LOCK TABLE celo_transaction_stats IN EXCLUSIVE MODE;
```

Comment:
Did you try other less restrictive modes like `SHARE ROW EXCLUSIVE`?

Reply:
I didn't actually, I saw that the transaction already has `SHARE` which doesn't prevent deadlocks and searched for a mode that enables only `select` statements + came upon the exclusive mode. `SHARE ROW EXCLUSIVE` would be the only other choice between share and exclusive.

Because the function can update both gas_used and tx count rows (i.e. the entire table) a full lock would be required